File: src/main/java/com/hjj/apiserver/domain/Bank.java
package com.hjj.apiserver.domain;

import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "tb_bank")
public class Bank extends BaseTimeEntity_Java {

    @RequiredArgsConstructor
    @Getter
    public enum BankType {
        BANK(020000, "은행"),
        SAVING_BANK(030300, "저축은행");

        private final int topFinGrpNo;
        private final String title;
    }

    @Id
    @Column(length = 20)
    private String finCoNo;

    @Column
    private String dclsMonth;
    @Column(length = 100)
    private String korCoNm;
    @Column
    private String dclsChrgMan;
    @Column
    private String hompUrl;
    @Column(length = 50)
    private String calTel;

    @Column
    @Enumerated(EnumType.STRING)
    private BankType bankType;

    @Column
    private int enable;


    @OneToMany(mappedBy = "bank", fetch = FetchType.LAZY)
    @Builder.Default
    private List<Deposit> deposits = new ArrayList<>();

}

File: src/main/kotlin/com/hjj/apiserver/domain/bank/Bank.kt
package com.hjj.apiserver.domain.bank

import com.hjj.apiserver.domain.BaseTimeEntity
import jakarta.persistence.*

@Entity
@Table(name = "tb_bank")
class Bank(
    dclsMonth: String? = null,
    korCoNm: String? = null,
    dclsChrgMan: String? = null,
    hompUrl: String? = null,
    calTel: String? = null,
    bankType: BankType,
    enable: Int = 1,
) : BaseTimeEntity() {
    @Id
    @Column(length = 20)
    val finCoNo: String? = null

    @Column(length = 20)
    var dclsMonth: String? = dclsMonth
        protected set

    @Column(length = 100)
    var korCoNm: String? = korCoNm
        protected set

    @Column
    var dclsChrgMan: String? = dclsChrgMan
        protected set

    @Column
    var hompUrl: String? = hompUrl
        protected set

    @Column(length = 50)
    var calTel: String? = calTel
        protected set

    @Column
    @Enumerated(EnumType.STRING)
    var bankType: BankType = bankType
        protected set

    @Column
    var enable: Int = enable
        protected set
}
